xml/System.Text.RegularExpressions/Regex.xml

```diff
@@ -3066,9 +3068,13 @@ Allows an <see cref="T:System.Object" /> to attempt to free resources and perfor
   
- You can optionally specify a starting position in the string by using the `startat` parameter. When the regular expression engine parses from left to right (the default), the match and the scan move rightward, starting at the character specified in `startat`. When the regular expression engine parses from right to left (when the regular expression pattern is constructed with the <xref:System.Text.RegularExpressions.RegexOptions.RightToLeft?displayProperty=nameWithType> option), the match and scan move in the opposite direction and begin with the character at `startat` -1. If you do not specify a starting position, the search begins at the default `startat` position. If the regular expression searches from left to right, the default `startat` position is at the left end of `input`; if it searches from right to left, the default `startat` position is at the right end of `input`.  
-  
- If you want to restrict a match so that it begins at a particular character position in the string and the regular expression engine does not scan the remainder of the string for a match, anchor the regular expression with a `\G` (at the left for a left-to-right pattern, or at the right for a right-to-left pattern). This restricts the match so it must start exactly at `startat`.  
+ You can optionally specify a starting position in the string by using the `startat` parameter. When the regular expression engine parses from left to right (the default), any matches starting before `startat` in the string are ignored. When the regular expression engine parses from right to left (when the regular expression pattern is constructed with the <xref:System.Text.RegularExpressions.RegexOptions.RightToLeft?displayProperty=nameWithType> option), the match and scan move in the opposite direction and begin with the character at `startat` - 1 and any matches ending after `startat` are ignored. If you do not specify a starting position, the search begins at the default `startat`. If the regular expression searches from left to right, the default `startat` position is at the left end of `input`; if it searches from right to left, the default `startat` position is at the right end of `input`.
+
+ Despite starting at `startat`, the index of any returned match will be relative to the start of the string.
+
+ Although the regular expression engine will not return any match starting before `startat` (or ending after `startat`, for a right-to-left pattern), it does not ignore the string before `startat` (after for a right-to-left pattern). This means that any anchors, such as `^` or `\A`, or zero-width assertions at the start of the pattern may or may not be satisfied depending on the content of the string before `startat` (after for a right-to-left pattern).
```

Comment:
```suggestion
 Although the regular expression engine doesn't return any match starting before `startat` (or ending after `startat`, for a right-to-left search), it doesn't ignore the string before `startat`. This means that any anchors, such as `^` or `\A`, or zero-width assertions at the start of the pattern may or may not be satisfied depending on the content of the string before `startat`.
```

Reply:
I think right-to-left only means that the search starts at the right end instead of the left end. The pattern itself is still matched from left to right.

Reply:
By "... (after for a RTL pattern)", do you mean it would be to the left of `startat`? I think before/after could be confusing here.

Reply:
No, to the right. Eg., given text "abcxxx" the pattern "abc(?=def)" and startat of 3 in a RTL match, it will start on the "c" and move left, but the match will fail because the "(?=def)" lookahead does not match the "xxx".

This is further complicated because when you pass "startat" to a RTL pattern, it starts at "startat - 1" (this is already documented) 

I'll think about how to write this more clearly. It's all a bit unfortunate, since 99.9% of people reading these docs don't care about RTL matching.

Reply:
Maybe we should just have a separate section for RTL searches and have all the info specific to that kind of search there, instead of cluttering up the other paragraphs where 99.9% of readers don't need to know about it.

Reply:
That would be my preference. I could even imagine removing any mention of RTL from all of the regex API docs except to link to one doc or section that says "but if you request RTL, these are all the things that change".

I don't have time to work on such a change though. What we have has sufficed for 20 years, I guess.